Innoactive/Streaming/ViveMediaDecoderStreamPlayer.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections;
+using HTC.UnityPlugin.Multimedia;
+using UnityEngine;
+
+namespace Innoactive.Hub.Media.Streaming
+{
+    /// <summary>
+    /// Implementation of <see cref="IVideoStreamPlayer"/> for the Vive Media Decoder plugin.
+    /// </summary>
+    [RequireComponent(typeof(ViveMediaDecoder))]
+    public class ViveMediaDecoderStreamPlayer : MonoBehaviour, IVideoStreamPlayer
+    {
+        protected ViveMediaDecoder MediaDecoder;
+
+        [SerializeField]
+        [Tooltip("Interval in seconds between playback attempts for an inactive player.")]
+        protected float PollingIntervalSeconds = 5;
+
+        protected WaitForSeconds PollingInterval;
+
+        /// <inheritdoc />
+        public event EventHandler<AspectRatioChangedEventArgs> AspectRatioChanged;
+
+        /// <inheritdoc />
+        public void SetStreamUrl(string url)
+        {
+            MediaDecoder.initDecoder(url);
+        }
+
+        /// <inheritdoc />
+        public void Play()
+        {
+            ViveMediaDecoder.DecoderState state = MediaDecoder.getDecoderState();
+
+            if (state == ViveMediaDecoder.DecoderState.INITIALIZED ||
+                state == ViveMediaDecoder.DecoderState.PAUSE ||
+                state == ViveMediaDecoder.DecoderState.STOP)
+            {
+                StartDecoding();
+            }
+            else
+            {
+                MediaDecoder.onInitComplete.AddListener(StartDecoding);
+            }
```

Comment:
How? If it's buffering it's already starting and nothing happens, if it's initializing it will start once initialize is complete. We could discuss about eof and seek frame, maybe. I don't really get the order of that enum...

Reply:
I checked both and you are right, for `Initializing` the stream will play automatically and for `Buffering` nothing would happen (apart of being registered on the listener). 

This last scenario would be the same for `Seek_Frame` and `EOF`.

Since the video player is meant to be used for video streaming I guess doesn't make much sense to validate when the video is over (`End of File`) but in the other hand, it wouldn't hurt to add it and that way we wouldn't limit functionality.

As you consider it :innocent:

Reply:
Added check for EOF.